lib/core/src/Cardano/Wallet/Primitive/CoinSelection/MA/RoundRobin.hs

```diff
@@ -1145,6 +1174,194 @@ makeChangeForCoin targets excess =
     weights :: NonEmpty Natural
     weights = coinToNatural <$> targets
 
+--------------------------------------------------------------------------------
+-- Equipartitioning
+--------------------------------------------------------------------------------
+
+-- An /equipartition/ of a value 'v' (of some type) is a /partition/ of that
+-- value into 'n' smaller values whose /sizes/ differ by no more than 1. The
+-- the notion of /size/ is dependent on the type of value 'v'.
+--
+-- In this section, equipartitions have the following properties:
+--
+--  1.  The length is observed:
+--      >>> length (equipartition v n) == n
+--
+--  2.  The sum is preserved:
+--      >>> sum (equipartition v n) == v
+--
+--  3.  Each resulting value is less than or equal to the original value:
+--      >>> all (`leq` v) (equipartition v n)
+--
+--  4.  The resultant list is sorted into ascending order when values are
+--      compared with the 'leq' function.
+--
+--------------------------------------------------------------------------------
+
+-- | Computes the equipartition of a coin into 'n' smaller coins.
+--
+equipartitionCoin
+    :: HasCallStack
+    => Coin
+    -- ^ The coin to be partitioned.
+    -> NonEmpty a
+    -- ^ Represents the number of portions in which to partition the coin.
+    -> NonEmpty Coin
+    -- ^ The partitioned coins.
+equipartitionCoin c =
+    -- Note: the natural-to-coin conversion is safe, as equipartitioning always
+    -- guarantees to produce values that are less than or equal to the original
+    -- value.
+    fmap unsafeNaturalToCoin . equipartitionNatural (coinToNatural c)
+
+-- | Computes the equipartition of a natural number into 'n' smaller numbers.
+--
+equipartitionNatural
+    :: HasCallStack
+    => Natural
+    -- ^ The natural number to be partitioned.
+    -> NonEmpty a
+    -- ^ Represents the number of portions in which to partition the number.
+    -> NonEmpty Natural
+    -- ^ The partitioned numbers.
+equipartitionNatural n count =
+    -- Note: due to the behaviour of the underlying partition algorithm, a
+    -- simple list reversal is enough to ensure that the resultant list is
+    -- sorted in ascending order.
+    NE.reverse $ unsafePartitionNatural n (1 <$ count)
+
+-- | Computes the equipartition of a token bundle into 'n' smaller bundles.
+--
+-- Each asset is partitioned independently.
+--
+equipartitionTokenBundle
+    :: HasCallStack
+    => TokenBundle
+    -- ^ The bundle to be partitioned.
+    -> NonEmpty a
+    -- ^ Represents the number of portions in which to partition the bundle.
+    -> NonEmpty TokenBundle
+    -- ^ The partitioned bundle.
+equipartitionTokenBundle (TokenBundle c m) count =
```

Comment:
This function shouldn't be needed, right? Only equipartitionTokenMap.

Reply:
> This function shouldn't be needed, right? Only equipartitionTokenMap.

You're right -- I've deleted it.

Fixed in 607ec07474ba813e9e7adc344c58c28b7f8f8fac